### PR TITLE
Add project method for nullspace projection

### LINKED ISSUE #5: Projection onto nullspace of matrix
We need a class that will faciliate the projection onto the nullspace of a matrix.

### PR DESCRIPTION
**Pull request by ruairimoran**
## Main Changes

- [x] Add `.project()` to DTensor
- [x] Test project method

## Associated Issues

- Closes #5

### CODE REVIEW

File: test/testTensor.cu [hunk 1]
```diff
@@ -920,5 +917,49 @@ void computeNullspaceTrivial(T epsilon) {
...
+    DTensor<T> x(vec, n);
+    DTensor<T> proj(x);
+    ns.project(proj);
+
+    // Testing that proj is indeed in ker A
+    DTensor<T> error(m, 1, 1, true);
+    error.addAB(A, proj);
+    EXPECT_TRUE(error.normF() < epsilon);
+
+    // Orthogonality test (other - p) † (p - x)
```
Comment (alphaville) [inline test/testTensor.cu:979]: I did some additional testing here

File: include/tensor.cuh [hunk 2]
```diff
@@ -221,6 +221,13 @@ public:
      */
     DTensor(const DTensor &other, size_t axis, size_t from, size_t to);
 
+    /**
+     * Frobenius dot product
+     * @param other other tensor of compatible dimensions
+     * @return value of Frobenius dot product
+     */
+    T dotF(const DTensor &other);
```
Comment (alphaville) [inline include/tensor.cuh]: I implemented this, but needs testing

Comment (ruairimoran) [reply include/tensor.cuh]: tested

**Review by alphaville — APPROVED** [review]
LGTM

**Pull request merged**